src/java.base/share/classes/jdk/internal/foreign/layout/ValueLayouts.java

```diff
@@ -159,20 +161,21 @@ static boolean isValidCarrier(Class<?> carrier) {
 
         @ForceInline
         public final VarHandle varHandle() {
-            final class VarHandleCache {
-                private static final Map<ValueLayout, VarHandle> HANDLE_MAP = new ConcurrentHashMap<>();
+            record VarHandleCache() implements Function<AbstractValueLayout<?>, VarHandle> {
```

Comment:
Can this cache be removed? The var handles created here should be "direct" -- meaning that we don't expect LayoutPath to add any adaptations on top of what's already done in the "raw" var handle returned by `Utils:makeRawSegmentViewVarHandle`?

Reply:
We have two caches, one in `ValueLayouts` and another in `Utils`. Should we remove both?

Reply:
I'm not convinced about the one in ValueLayouts *e.g. where this comment was added) -- if you agree that the other cache should already handle this -- then maybe we can only keep one

Reply:
Indeed, the other cache already handles this.

Reply:
Done in latest update.